java/client/src/main/java/com/youtube/vitess/client/cursor/StreamCursor.java

```diff
@@ -90,4 +91,31 @@ protected Row getCurrentRow() throws SQLException {
     }
     return row;
   }
+
+  /**
+   * Fetches the next {@link QueryResult} from the stream.
+   *
+   * <p>Whereas the public {@link #next()} method advances the {@link Cursor} state to the next
+   * {@link Row}, this method advances the internal state (namely {@link #rowIterator}) to the next
```

Comment:
I guess it's ambiguous. What I meant by "advancing the internal state" was, "assigning a new value to the rowIterator field". But you're right that this also entails "telling streamIterator to advance its own internal state". I'll see if I can clarify the comment.